src/docs/arch-eviction.dox

```diff
@@ -1,15 +1,108 @@
+end, reconciliation flushes the new disk image to the data file on the disk and later eviction
+removes the old page from memory.
+The history store table is the internal table for WiredTiger to store historical values for all the
+user-created tables. Each entry in the history store represents a historical value of a key in the
+user-created tables. The pages in the history store table need to be evicted and loaded back to memory
+as well. Different from the user-created tables, all the content on the history store page is committed
+as the historical values it stores must all be committed, and each entry only has one value and a
+deletion marking the expiration of the entry on its update chain. Therefore, history store pages
+should always be clean after reconciliation and can be evicted from memory. Besides, only having
+one committed value for each key in the history store prevents eviction from moving content already
+in the history store recursively to the history store when evicting a history store page because we
+only move the values after the first committed value to the history store.
+
+@subsection update_restore_eviction_1 Update restore eviction
+Update restore eviction largely overlaps with history store eviction. The difference is that there
+are uncommitted changes on the page and WiredTiger cannot write them to the disk image. Instead, besides
+writing the disk image to the disk, eviction also keeps the disk image in memory and restores the
+uncommitted changes to the new disk image.
+Since usually there is only a subset of keys that have uncommitted changes, eviction only needs to
+restore those keys that are dirty after reconciliation. Apart from that, eviction may split a page
+into multiple pages and some of them may be clean after reconciliation even though the old page before
+split has uncommitted changes. Therefore, WiredTiger tracks whether it needs to do update restore eviction
+at the key level and the page level.
+At the key level, the key needs to be restored if it has uncommitted updates. For each new page, restoration
+is required if any of the keys on the new page needs to be restored. Once eviction decides to restore a
+page, it will make a copy of the disk image in memory during reconciliation.
+For the restored keys, eviction also frees all the values that have been written to the disk image or
+moved to the history store as there is no need to duplicate them on the update chains. For the clean
+keys that are not restored, their whole update chains are removed from memory along with the old disk image.
+
+Eviction triggers
+
+WiredTiger provides several configuration options for tuning how pages are evicted from the cache.
+
+WiredTiger eviction tuning options can be configured when first opening
+a database via ::wiredtiger_open, or changed after open with
+WT_CONNECTION::reconfigure.
+
+The \c eviction_target configuration value (default 80%) is the level at
```

Comment:
I am not sure if we should mention the default values. Though we have not changed defaults in a long time, if we do in the future, we might forget to update here in the guide.

Reply:
Removed the default values.